.github/workflows/build.yml

```diff
@@ -246,3 +245,50 @@ jobs:
           dev_tags: |
             hashicorppreview/${{ env.repo }}:${{ env.version }}
             hashicorppreview/${{ env.repo }}:${{ env.version }}-${{ github.sha }}
+
+  upload-dev-docker:
+    name: Upload dev image (hashicorpdev/consul-ecs:<commit>)
+    environment: dockerhub/hashicorpdev
+    needs:
+      - get-product-version
+      - build-docker-default
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: ["amd64"]
+    env:
+      repo: ${{ github.event.repository.name }}
+      version: ${{ needs.get-product-version.outputs.product-version }}
+      target: release-default
+      git-short-sha: ${{ needs.get-product-version.outputs.git-short-sha }}
+    steps:
+      - name: download image artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ env.repo }}_${{ env.target }}_linux_${{ matrix.arch }}_${{ env.version }}_${{ github.sha }}.docker.dev.tar
+      - name: docker push
+        shell: bash
+        run: |
+          TAG=hashicorpdev/${{ env.repo }}:${{ env.git-short-sha }}
+          echo "==> Load docker image from tar archive"
+          docker load -i "${{ env.repo }}_${{ env.target }}_linux_${{ matrix.arch }}_${{ env.version }}_${{ github.sha }}.docker.dev.tar"
+
+          echo "==> Tag docker image $TAG"
+          docker tag "hashicorppreview/${{ env.repo }}:${{ env.version }}-${{ github.sha }}" "$TAG"
+
+          echo "==> Docker login"
+          echo ${{ secrets.DOCKER_PASS }} | docker login -u=${{ secrets.DOCKER_USER }} --password-stdin
```

Comment:
In general we're trying to rely less heavily on GitHub secrets and instead entrust CRT common workflows running on our private infra to access needed secrets from Vault to perform work like this. I'm curious if we can better support your use case so we can avoid the need for this?